### PR TITLE
Added ability to define skip values in csv parser

### LINKED ISSUE #6436: Add the ability to replace or ignore values in the CSV parser
## Feature Request

Opening a feature request kicks off a discussion.

### Proposal:
When trying to parse data from https://www.ndbc.noaa.gov/data/latest_obs/latest_obs.txt , they substitute `MM` in for fields that are missing a value. This data cannot be parsed by the CSV parser because if it hits that `MM`, then it throws a field conversion error.

It would be nice to be able to define a set of values to "ignore" when parsing CSV.

### Current behavior:
There is no way to configure ignoring missing values.

### Desired behavior:
A way to ignore specific values for parsing csv data where a value is substituted in for a missing value, such as in the link above.

### Use case: 
This would allow me to parse additional CSV data without needed to pre-process it or write a custom script.

### PR DESCRIPTION
**Pull request by helenosheaa**
Closes #6436 

Adds the ability to define an array of string skip values to the csv parser, such as an empty string, where the value matches the provided skip value it will skip the field entirely. Addresses the issue of ignoring missing values

### CODE REVIEW

File: plugins/parsers/csv/parser.go [hunk 1]
```diff
@@ -197,6 +198,15 @@ outer:
 				value = strings.Trim(value, " ")
 			}
 
+			// don't record fields where the value matches a skip value
+			if len(p.SkipValues) > 0 {
```
Comment (Jagularr) [inline plugins/parsers/csv/parser.go:202]: Is checking the length necessary?

**Review by Jagularr — COMMENTED** [review]
Looks good to me

File: plugins/parsers/csv/parser.go (hunk 1 above)
Comment (helenosheaa) [reply plugins/parsers/csv/parser.go:202]: You're right it's redundant, I'll take it out

File: plugins/parsers/csv/parser.go [hunk 2]
```diff
@@ -197,6 +198,14 @@ outer:
 				value = strings.Trim(value, " ")
 			}
 
+			// don't record fields where the value matches a skip value
+			for _, s := range p.SkipValues {
+				fmt.Println("in here")
```
Comment (reimda) [inline plugins/parsers/csv/parser.go:203]: Remove debugging println

**Review by reimda — CHANGES_REQUESTED** [review]

File: plugins/parsers/csv/parser.go (hunk 2 above)
Comment (helenosheaa) [reply plugins/parsers/csv/parser.go]: Done, I'll make sure I check more carefully for println in the future before I commit!

Comment (reimda) [reply plugins/parsers/csv/parser.go]: No problem, they sneak in from time to time for everybody.  That's what reviews are for.

**Review by reimda — APPROVED** [review]

**Pull request merged**